torch/_inductor/codegen/halide.py

```diff
@@ -389,19 +390,52 @@ def bitwise_right_shift(a, b):
 
     @staticmethod
     def rand(seed, offset):
-        raise Unsupported("rand")
+        return f"hl.random_float({seed} ^ {offset})"
 
     @staticmethod
     def randn(seed, offset):
-        raise Unsupported("rand")
+        # Box-Muller transform
+        return ops.mul(
+            ops.sqrt(
+                ops.mul(
+                    ops.constant(-2, torch.float32), ops.log(ops.rand(seed, offset))
+                )
+            ),
+            ops.cos(
```

Comment:
Not sure if having a randn2x that also returns the other random number computed with ops.sin can be more efficient. But like triton's rand4x , the tricky part is how to integrate into inductor so we can properly codegen then